sources/representation/qt/mainform.cpp

```diff
@@ -178,43 +184,95 @@ void MainForm::startGameLoop(int id, QString map)
     qint64 max_process_time = 0;
-
-    while (true)
+    
+    if (!cap)
     {
-        ++fps_counter;
-
-        process_performance.start();
-        GetRepresentation().Process();
-        GetScreen().Swap();
-        qint64 process_time = process_performance.nsecsElapsed();
-        if (process_time > max_process_time)
+        while (true)
         {
-            max_process_time = process_time;
-        }
+            ++fps_counter;
 
-        if (isHidden())
-        {
-            break;
-        }
+            process_performance.start();
+            GetRepresentation().Process();
+            GetScreen().Swap();
+            qint64 process_time = process_performance.nsecsElapsed();
+            if (process_time > max_process_time)
+            {
+                max_process_time = process_time;
+            }
 
-        if (fps_timer.elapsed() > 1000)
+            if (isHidden())
+            {
+                break;
+            }
+
+            if (fps_timer.elapsed() > 1000)
+            {
+                addSystemText("FPS", "FPS: " + QString::number(fps_counter));
+                addSystemText(
+                    "{Perf}Represent",
+                    "Represent max: "
+                    + QString::number(max_process_time / 1e6)
+                    + " ms");
+                qint64 mutex_ns = GetRepresentation().GetPerformance().mutex_ns;
+                addSystemText(
+                    "{Perf}RepresentMutex",
+                    "Represent mutex lock max: "
+                    + QString::number(mutex_ns / 1e6)
+                    + " ms");
+                GetRepresentation().ResetPerformance();
+                max_process_time = 0;
+                fps_timer.restart();
+                fps_counter = 0;
+            }
+        }
+    }
+    else
+    {
+        int delta_time = 0;
+        float fps_limit = 1000 / cap;
+        QTime delta;
+        delta.start();
+        while (true)
         {
-            addSystemText("FPS", "FPS: " + QString::number(fps_counter));
-            addSystemText(
-                "{Perf}Represent",
-                  "Represent max: "
-                + QString::number(max_process_time / 1e6)
-                + " ms");
-            qint64 mutex_ns = GetRepresentation().GetPerformance().mutex_ns;
-            addSystemText(
-                "{Perf}RepresentMutex",
-                  "Represent mutex lock max: "
-                + QString::number(mutex_ns / 1e6)
-                + " ms");
-            GetRepresentation().ResetPerformance();
-            max_process_time = 0;
-            fps_timer.restart();
-            fps_counter = 0;
+            process_performance.start();
+            delta_time += delta.restart();
+            if (delta_time > fps_limit)
+            {
+                delta_time -= fps_limit;
+                ++fps_counter;
+                GetRepresentation().Process();
+                GetScreen().Swap();
+            }
```

Comment:
The main purpose of the issue is to stop the representation thread from 100% loading of a processor core.
You should calculate a time which one frame should consume (`time_per_frame_ms = 1000 / cap_fps`) and then the representation thread should sleep the rest of this time.
Example: 
```
-max_fps=10
time_per_frame_ms = 100;
process_time_ms = 35;
sleep_time_ms = time_per_frame_ms - process_time_ms = 65;
```
So that means you should sleep _after_ the representation processing for 65 ms.
If `max_fps` is undefined then you should just make `time_per_frame_ms` zero and everything will work without changes.